File: algorithms/2_medium/M/02285_maximum_total_importance_of_roads/solution.c
static int cmp(const void *a_ptr, const void *b_ptr) {
    const int a = *(const int*)a_ptr;
    const int b = *(const int*)b_ptr;

    return (a > b) - (a < b);
}

long long maximumImportance(int n, int** roads, int roads_rows, int* roads_cols_data) {
    int *restrict deg = calloc((size_t)n, sizeof(*deg));
    if (deg == NULL) { return -1; }

    for (int i = 0; i < roads_rows; i += 1) {
        const int u = roads[i][0];
        const int v = roads[i][1];
        deg[u] += 1;
        deg[v] += 1;
    }

    qsort(deg, (size_t)n, sizeof(*deg), cmp);

    int64_t res = 0;
    for (int i = 0; i < n; i += 1) {
        const int64_t imp = i + 1;
        const int64_t d = deg[i];
        res += d * imp;
    }

    free(deg);
    return res;
}

File: algorithms/2_medium/M/02285_maximum_total_importance_of_roads/solution.rs
impl Solution {
    pub fn maximum_importance(n: i32, roads: Vec<Vec<i32>>) -> i64 {
        let mut deg = vec![0_i32; n as usize];
        for r in roads {
            let (u, v) = (r[0], r[1]);
            deg[u as usize] += 1;
            deg[v as usize] += 1;
        }

        deg.sort_unstable();

        let mut res: i64 = 0;
        for (i, d) in deg.into_iter().enumerate() {
            let imp = (i + 1) as i64;
            let d = d as i64;
            res += d * imp;
        }

        res
    }
}
